### PR TITLE
Implement API generation. Resolves #39

### LINKED ISSUE #39: API Documentation
Needs proper API documentation.  This might be included as a section of the `README.md` file.

Things to document: 

- [x] BScript class 
  - [x] Constructor
  - [x] Instance Properties
    - [x] `tokens`
    - [x] `hasAddress`
    - [x] `scriptType`
    - [x] `redeemScript`
  - [x] Instance Methods
    - [x] `toRaw`
    - [x] `toAsm`
    - [x] `toAddress`
  - [x] Static Methods
    - [x] `fromRaw`
    - [x] `fromAsm`
    - [x] `fromAddress`
- [x] Token class
  - [x] Constructor
  - [x] Instance Properties
    - [x] `type`
    - [x] `value`
    - [x] `startIndex`
    - [x] `endIndex`
  - [x] Instance Methods
    - [x] `toScript`
    - [x] `toAsm`
    - [x] `toString` (alias for `toAsm`)
  - [x] Static Properties
    - [x] Token Types: `LITERAL`, `OPCODE`, `PLACEHOLDER`
  - [x] Static Methods
    - [x] Constructor Helpers: `literal`, `opcode`, `placeholder`
- [x] Convenience Methods
  - [x] `rawToAsm`
  - [x] `rawToAddress`
  - [x] `asmToRaw`
  - [x] `asmToAddress`
  - [x] `addressToRaw`
  - [x] `addressToAsm`
- [x] Opcode Helpers
  - [x] `opcodeForWord`
  - [x] `wordForOpcode`
  - [x] `opcodeIsValid`
  - [x] `wordIsValid`
  - [x] `descriptionForOpcode`
  - [x] `descriptionForWord`
  - [x] `inputDescriptionForOpcode`
  - [x] `inputDescriptionForWord`
  - [x] `outputDescriptionForOpcode`
  - [x] `outputDescriptionForWord`
  - [x] `opcodeIsDisabled`
  - [x] `wordIsDisabled`
- [x] Options Struct
  - [x] `encoding` (for raw script buffer conversion)
  - [x] `literalStyle` (for asm stringification)
  - [x] `opcodeStyle` (for asm stringification)
  - [x] `terms` (object of overrides & additions for term opcodes)
  - [x] `opcodes (object of overrides for opcode terms, inverse of terms)`
  - [x] `pubKeyHash`
  - [x] `scriptHash`
  - [x] `bech32` (the prefix, i.e. `'bc'`)
  - [x] `allowPlaceHolders` (for asm parser)

### PR DESCRIPTION
**Pull request by ghost**
It is largely a matter of opinion whether or not one believes this style of documentation improves the quality of the code.  As such, I have only implemented the documentation for the main module and would like to get some feedback before I continue.  Do you think this is the correct path to take on this, or do you think there is a better one?

Don't forget to run `npm install`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,3 +80,11 @@ true
 > bscript.opcodes.outputDescriptionForWord('OP_HASH160')
 'hash'
 ```
+
+### API Docs
+
+You can view the the API docs for this project by running `npm run docs:server` in the project directory and then accessing
```
Comment (JBaczuk) [inline README.md]: should be `npm run docs:serve`

Comment (JBaczuk) [inline README.md]:
```suggestion
You can view the the API docs for this project by running `npm run docs:serve` in the project directory and then accessing
```

**Comment by JBaczuk** [comment]
I really like this, I'm used to either reading a long README on a git repo, or else a link to the docuentation. So, I think we should create a github.io page for this once it is released, and link to that in the README.

**Comment by wardlem** [comment]
> I really like this, I'm used to either reading a long README on a git repo, or else a link to the docuentation. So, I think we should create a github.io page for this once it is released, and link to that in the README.

Oh great! I will continue working on this then.

**Comment by wardlem** [comment]
@JBaczuk I got most of this done. Waiting on the resolution of #46 and #34 before taking this out of WIP status.

**Comment by wardlem** [comment]
@JBaczuk This is out of WIP status and is ready for review.

**Comment by JBaczuk** [comment]
rebase?

**Review by JBaczuk — APPROVED** [review]
Freaking awesome man, I think this helps the quality of the code as an open source project. I think it just needs to be rebased.

**Pull request merged**